maven-plugin/src/main/java/org/jvnet/localizer/Generator.java

```diff
@@ -65,45 +78,92 @@ protected void generateImpl(String className, Properties props) throws Assertion
+            c.javadoc().add("Generated localization support class.");
 
             // [RESULT]
             // private static final ResourceBundleHolder holder = BundleHolder.get(Messages.class);
 
-            JVar holder = c.field(JMod.PRIVATE | JMod.STATIC | JMod.FINAL, ResourceBundleHolder.class, "holder",
+            JFieldVar holder = c.field(JMod.PRIVATE | JMod.STATIC | JMod.FINAL, ResourceBundleHolder.class, "holder",
                     cm.ref(ResourceBundleHolder.class).staticInvoke("get").arg(c.dotclass()) );
+            holder.javadoc().add("The resource bundle reference");
 
 
             for (Entry<Object,Object> e : props.entrySet()) {
                 String key = e.getKey().toString();
                 String value = e.getValue().toString();
 
-                int n = countArgs(value);
+                final Format[] formats = new MessageFormat(value).getFormatsByArgumentIndex();
+                Map<String,String> params = new HashMap<String, String>();
 
                 // generate the default format method
                 List<JVar> args = new ArrayList<JVar>();
                 JMethod m = c.method(JMod.PUBLIC | JMod.STATIC, cm.ref(String.class), toJavaIdentifier(key));
-                for( int i=1; i<=n; i++ )
-                    args.add(m.param(Object.class,"arg"+i));
+                for( int i=0; i<formats.length; i++ ) {
+                    String argName = String.format("arg%d", i);
+                    args.add(m.param(inferType(formats[i]), argName));
+                    if (formats[i] instanceof NumberFormat) {
+                        params.put(argName, String.format("%s format parameter, {@code {%d}}, a number.",
+                                positionalName(i), i));
+                    } else if (formats[i] instanceof DateFormat) {
+                        params.put(argName,
+                                String.format("%s format parameter, {@code {%d}}, a {@link Date}.",
+                                        positionalName(i), i));
+                    } else {
+                        params.put(argName, String.format(
+                                "%s format parameter, {@code {%d}}, as {@link Object#toString()}.",
+                                positionalName(i), i));
+                    }
+                }
 
                 JInvocation inv = holder.invoke("format").arg(key);
                 for (JVar arg : args)
                     inv.arg(arg);
                 m.body()._return(inv);
 
-                m.javadoc().add(escape(value));
+                m.javadoc().add(WordUtils.wrap(String.format(
+                        "Returns the formatted message string for key %s from the bundle using the {@link org.jvnet"
+                                + ".localizer.LocaleProvider#getLocale()}, in the default {@link java.util.Locale} the"
+                                + " message string is %s.",
+                        code(key), code(value)), 70));
```

Comment:
I want the key also as sometimes you can have two keys folding down to the same, e.g. `Foo.DisplayName` and `Foo_DisplayName`